### ISSUE TITLE
RFE: support  specify files to get back from the environment

### ISSUE THREAD

**Issue by bgoncalv** [psss/tmt#103]
For example how to get all *.log and *.yml from directory the script was run.

It should not fail if some file doesn't exist.

**Comment by thrix** [comment]
I will try to define this!

**Comment by pvalena** [comment]
Get back to where?

Note, that from the implementation side, everything from guest, on which the tests run, should be synced back in `/var/tmp/tmt/run-$NUM/`, which is the run ID and workdir at the same time.

**Comment by bgoncalv** [comment]
Let's say I have a template that will execute an script in a container. This script will create few files in the container and I'd like to be able to get this files from the container.

**Comment by pvalena** [comment]
> Let's say I have a template that will execute an script in a container. This script will create few files in the container and I'd like to be able to get this files from the container.

Maybe I'm looking at this wrong, but I think my question is still standing- where to do you want to copy the files a test created? There is a reporting step, which (when implemented) should handle the results, f.e. send / upload results somewhere etc. (when running as CI, or gating test). If you mean on your local host, when you want to check the state while developing your tests, it's all all in `/var/tmp/tmt/run-...` (or other folder you've specified as `--id`).

Is it more clear now, or I've missed something?

**Comment by bgoncalv** [comment]
Right, I believe it should be part of reporting step, together with the results it should also store some extra files that I define.

When run on localhost, what do you mean by all is in `/var/tmp/tmt/run-...`? Does it mean if my script download a bunch of packages, or large files like qcow2, that I don't need them to be stored they will still be copied to `/var/tmp/tmt/run-...`?

**Comment by pvalena** [comment]
> When run on localhost, what do you mean by all is in `/var/tmp/tmt/run-...`? Does it mean if my script download a bunch of packages, or large files like qcow2, that I don't need them to be stored they will still be copied to `/var/tmp/tmt/run-...`?

(Not sure what you mean by `localhost`, as we have a `localhost` provision implementation, which does not use VMs or containers and simply runs everything as the `tmt`'s user. When something's downloaded, then it will be there, off course.)

When you use default provision (or anything that's not `localhost`), then, at this point, also **yes**. 
  - We rsync everything in the _workdir_ to-and-from a guest, as it's the simplest implementation to begin with.
  - If it's not in tmt's workdir (f.e. some tmpdir), it will not be rsynced. It can be copied to the workdir _on demand_ (from inside tmt), but there's no API yet.
  - It's still early in the tools' development process and the syncing-back-and-forth will be worked on and optimized. 

Also containers are not yet supported, but we'll get to it. Currently we support, that I know of, only VMs and remote-guests _connected_ via ssh. They all use the same syncing implementation.

**Comment by thrix** [comment]
ah right, everything is already "archived" in the run directory, so I guess there is not much to define here in spec for now, moving myself out of the assignement

**Comment by lukaszachy** [comment]
Artifact field (once implemented) should be the way how to specify which files/directories should be pulled from guest.

**Comment by psss** [comment]
I believe this has been covered by the `TMT_TEST_DATA` variable:

    TMT_TEST_DATA
        Path to the directory where test can store logs and other
        artifacts generated during its execution. These will be
        pulled back from the guest and available for inspection
        after the test execution is finished.

It would be good to have at least a small section in the guide which would mention this to make it obvious for new users.
